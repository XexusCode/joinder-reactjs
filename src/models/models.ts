export interface UserEventObject {
  username: string;
  uid: string;
  color: string;
  rank: number;
}

export interface UserAuth {
  uid: string;
  username: string;
}

export interface EventObject {
  id?: string;
  title: string;
  startDate: string;
  endDate: string;
  location: any;
  nmax: number;
  img: string;
  description?: string;
  userEvents: Array<UserEventObject>;
  todos: Array<TodoObject>;
  comments: Array<CommentObject>;
}

export interface TodoObject {
  id: number;
  text: string;
}
export interface CommentObject {
  id: number;
  userEventUsername: string;
  text: string;
  date: string;
}
