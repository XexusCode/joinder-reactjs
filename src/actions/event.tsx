import { fetchApi } from "../helpers/fetch";
import { types } from "../types/types";
import { dispatchAction, iEvents } from "../interfaces/interfaces";
import { Dispatch } from "react";

export const startLoadEvents = (uid: string) => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    const resp = await fetchApi("loadevents", "GET");
    const body = await resp.json();
    if (body.success) {
      dispatch(loadEvents(body.data));
    }
  };
};

// export const startAddUserEvent = (uid: string, idEvent: string) => {
//   return async (dispatch: any) => {
//     const resp = await fetchApi("adduser", "GET");
//     const body = await resp.json();
//
//     if (body.success) {
//       dispatch(updateActiveEvent(body.data));
//     }
//   };
// };

export const startAddNewEvent = (event: iEvents) => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    const resp = await fetchApi("addevent", "GET");
    const body = await resp.json();

    if (body.success) {
      dispatch(addEvent(body.data));
      dispatch(updateActiveEvent(body.data));
    }
  };
};

export const startDeleteEvent = () => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    const resp = await fetchApi("deleteevent", "GET");
    const body = await resp.json();
    if (body.success) {
      dispatch(deleteEvent());
    }
  };
};

export const startJoinNewEvent = (idEvent: string, uid: string) => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    const resp = await fetchApi("joinevent", "GET");
    const body = await resp.json(); // Usuario aÃ±adido a evento (idEvent)

    if (body.success) {
      const event: iEvents = { ...body.data };
      dispatch(addEvent(event));
    }
  };
};

export const startUpdateEvent = (event: iEvents) => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    const resp = await fetchApi("updateevent", "GET");
    const body = await resp.json();

    if (body.success) {
      dispatch(updateActiveEvent(body.data));
    }
  };
};

export const startLogoutEvents = () => {
  return async (dispatch: Dispatch<dispatchAction>) => {
    dispatch(logoutEvents());
  };
};
export const loadEvents = (events: Array<iEvents>) => ({
  type: types.eventsLoad,
  payload: events,
});

export const setActiveEvent = (event: iEvents) => ({
  type: types.eventSetActive,
  payload: event,
});

export const updateActiveEvent = (event: iEvents) => ({
  type: types.eventUpdateActive,
  payload: event,
});

export const addEvent = (event: iEvents) => ({
  type: types.eventAddNew,
  payload: event,
});

export const deleteEvent = () => ({
  type: types.eventDelete,
});

export const logoutEvents = () => ({
  type: types.eventLogout,
});
