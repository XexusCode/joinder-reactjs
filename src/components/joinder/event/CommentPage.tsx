import React, { useState } from "react";

const CommentPage = () => {
  const [loading] = useState(false);

  const loadingSpin = loading ? "App-logo Spin" : "App-logo";

  return (
    <>
      <div className="App container bg-light shadow">
        <header className="App-header">
          <img
            src="https://static1.abc.es/media/tecnologia/2018/09/25/google-chrome-0-kCD--620x349@abc.jpg"
            className={loadingSpin}
            alt="logo"
          />
          <h1 className="App-title">
            React Comments
            <span className="px-2" role="img" aria-label="Chat">
              ðŸ’¬
            </span>
          </h1>
        </header>

        <div className="row">
          <div className="col-4  pt-3 border-right">
            <h6>Say something about React</h6>
            {/* Comment Form Component */}
          </div>
          <div className="col-8  pt-3 bg-white">
            {/* Comment List Component */}
          </div>
        </div>
      </div>
    </>
  );
};

export default CommentPage;
