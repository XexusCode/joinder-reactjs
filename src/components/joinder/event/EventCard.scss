.event{
  color: #000000;



  &:hover{
    background-color: darken(#7cbeec,20%);
    text-decoration: lavender;
    text-underline: none;


  }
}
.fondo{
  background-color: #7cbeec;
}

.badge{
  font-size: larger;
}
.nmax{

  padding-top: 10px;
}
hr{
  border: none;
}
.easy-edit-not-allowed{
  &:hover{
    cursor: auto;

  }
}