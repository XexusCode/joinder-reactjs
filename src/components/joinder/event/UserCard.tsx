import { UserObjects } from "../../../models/models";
import React from "react";
import { AiFillStar, CgTrash, FaCrown, RiUserFill } from "react-icons/all";
import { useUser } from "../../../modules/activeEvent/hooks/useUser";

interface UserCardParams {
  user: UserObjects;
  handleKickOut: () => void;
  handleRankUp: () => void;
}

export const UserCard = ({
  user,
  handleKickOut,
  handleRankUp,
}: UserCardParams) => {
  const { uid, rank } = useUser();

  return (
    <div className="pb-3">
      <span>
        <RiUserFill size={28} color={user.color} /> {user.username}{" "}
        {user.rank === 0 || user.rank === 1 ? (
          <AiFillStar color={user.rank === 0 ? "gold" : "#00aae4"} />
        ) : (
          <span />
        )}
        {rank <= 2 && uid != user.uid ? (
          <span className="pl-3">
            <span onClick={handleRankUp} className="crown">
              <FaCrown />
            </span>{" "}
            <span onClick={handleKickOut} className="trash">
              <CgTrash />
            </span>{" "}
          </span>
        ) : (
          <span />
        )}
        {uid === user.uid ? <span className="pl-3">[YOU]</span> : <div />}
      </span>
    </div>
  );
};
