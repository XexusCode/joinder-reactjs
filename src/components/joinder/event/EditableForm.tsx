import React from "react";
// @ts-ignore
import EasyEdit from "react-easy-edit";
import { GoCheck, GoX } from "react-icons/all";
import "./EventCard.scss";
import { ActiveEventEditableParams } from "../../../modules/activeEvent/ActiveEventParams";

export const EditableForm = ({
  editable,
  edit,
  handleSaveValue,
}: ActiveEventEditableParams) => {
  return (
    <EasyEdit
      className="easy-edit-not-allowed"
      type="text"
      value={editable}
      onSave={handleSaveValue}
      onCancel={() => {}}
      allowEdit={edit}
      saveButtonLabel={<GoCheck />}
      cancelButtonLabel={<GoX />}
      attributes={{ name: "awesome-input", id: 1 }}
    />
  );
};
