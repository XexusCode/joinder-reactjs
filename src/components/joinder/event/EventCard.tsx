import React from "react";
import "./EventCard.scss";
import { Badge, Col, Container, Image, Row } from "react-bootstrap";
import { Link } from "react-router-dom";
import { EventObject } from "../../../interfaces/interfaces";
export const EventCard = (event: EventObject) => {
  return (
    <Link style={{ textDecoration: "none" }} to={`./evento/${event.idevent}`}>
      <Col>
        <hr />
        <Row>
          <Container className="fondo event rounded ">
            <Col>
              <Row>
                <Col md={5}>
                  <div className="card-body  ">
                    <Image
                      src="http://placeimg.com/100/100/food"
                      roundedCircle
                    />
                  </div>
                </Col>
                <Col md={5}>
                  <div className="card-title  ">
                    <h3>{event.name}</h3>
                    <h6>Localizacion: {event.location}</h6>
                  </div>
                  <Row>
                    <hr />{" "}
                  </Row>
                  <Row md={10}>
                    <Badge variant="success">{event.start_date}</Badge> <hr />
                    <Badge variant="danger">{event.end_date}</Badge>
                  </Row>
                </Col>
                <Col md={1}>
                  <Row md={2}>
                    <div className="nmax">
                      {event.nmax >= event.users.length ? (
                        <h5 style={{ color: "green" }}>
                          {event.users.length}/{event.nmax}{" "}
                        </h5>
                      ) : (
                        <h5 style={{ color: "red" }}>
                          {event.users.length}/{event.nmax}{" "}
                        </h5>
                      )}
                    </div>
                  </Row>
                </Col>
              </Row>
            </Col>
          </Container>
        </Row>
      </Col>
    </Link>
  );
};
