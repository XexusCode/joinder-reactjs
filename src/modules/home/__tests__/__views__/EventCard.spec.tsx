import { render, cleanup, fireEvent } from "@testing-library/react";
import React from "react";
import { EventObject } from "../../../../models/models";
import { Router } from "react-router-dom";
import history from "../../../../routing/history";
import { EventCard } from "../../../../components/joinder/event/EventCard";

describe("test for EventCard", function () {
  const handleActiveEventMock = jest.fn();
  afterEach(cleanup);

  it("should have a event created", function () {
    const eventtest: EventObject = {
      comments: [],
      endDate: "",
      img: "",
      location: "",
      nmax: 0,
      startDate: "",
      title: "TEST",
      password: "e",
      todos: [],
      userEvents: [],
    };
    const { getByTestId } = render(
      <Router history={history}>
        <EventCard
          event={eventtest}
          handleActiveEvent={handleActiveEventMock}
        />
      </Router>
    );

    fireEvent.click(getByTestId("test-title"));

    expect(getByTestId("test-title")).toHaveTextContent("TEST");
    expect(handleActiveEventMock).toHaveBeenCalled();
  });


});
