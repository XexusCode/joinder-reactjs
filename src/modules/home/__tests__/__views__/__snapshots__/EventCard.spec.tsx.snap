// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for EventCard should take a snapshot from the component 1`] = `[Function]`;
  nbv mnbv
